op-defender/psp_executor/defender.go

```diff
@@ -0,0 +1,290 @@
+}
+
+// NewAPI creates a new HTTP API Server for the PSP Executor and starts listening on the specified port from the args passed.
+func NewDefender(ctx context.Context, log log.Logger, m metrics.Factory, cfg CLIConfig) (*Defender, error) {
+	// Set the route and handler function for the `/api/psp_execution` endpoint.
+	router := mux.NewRouter()
+	router.HandleFunc("/api/psp_execution", handlePost).Methods("POST")
+	l1client, err := GetTheL1Client()
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch l1 RPC: %w", err)
+	}
+
+	if cfg.portapi == "" {
+		return nil, fmt.Errorf("port.api is not set.")
+	}
+
+	defender := &Defender{
+		log:      log,
+		l1Client: l1client,
+		port:     cfg.portapi,
+		router:   router,
+	}
+
+	defender.log.Info("Starting HTTP JSON API PSP Execution server...", "port", defender.port)
+	return defender, nil
+}
+
+// FetchAndExecute() will fetch the privatekey, and correct PSP from GCP and execute it on the correct chain.
+func FetchAndExecute() {
+	//1. Fetch the privatekey of the account in GCP secret Manager.
+	privatekey, err := FetchPrivateKeyInGcp()
+	if err != nil {
+		log.Crit("Failed to fetch the privatekey from GCP secret manager: %v", err.Error())
+	}
+	// 2. Fetch the CORRECT Nounce PSP in the GCP secret Manager and return the data to execute.
+	superchainconfig_address, safe_address, data, err := FetchPSPInGCP()
+	if err != nil {
+		log.Crit("Failed to fetch the PSP from GCP secret manager: %v", err.Error())
+	}
+
+	// 3. Get the L1 client and ensure RPC is correct.
+	l1client, err := GetTheL1Client()
+	if err != nil {
+		log.Crit("Failed to get the L1client: %v", err.Error())
+	}
+
+	// 4. Execute the PSP on the chain.
+	ctx := context.Background() //TODO: Check if we really do need to context if yes we will keep it otherwise we will remove this.
+	PspExecutionOnChain(ctx, l1client, superchainconfig_address, privatekey, safe_address, data)
+}
+
+// GetTheL1Client() will return the L1 client based on the RPC provided in the config and ensure that the RPC is not production one.
+func GetTheL1Client() (*ethclient.Client, error) {
+	client, err := ethclient.Dial(LocalhostRPC) //Need to change this to the correct RPC (mainnet or sepolia) but for now hardcoded to localhost.
+	if LocalhostRPC != "http://localhost:8545" {
+		log.Warn("This is not the RPC localhost are you sure you want to continue (yes/no)")
+		var response string
+		fmt.Scanln(&response)
+		if response != "yes" {
+			log.Crit("Not yes, We Exiting the program.")
+		}
+	}
+	if err != nil {
+		log.Crit("Failed to connect to the Ethereum client: %v", err.Error())
+	}
+	return client, nil
+}
+
+// FetchPrivateKey() will fetch the privatekey of the account that will execute the pause (from the GCP secret manager).
+func FetchPrivateKeyInGcp() (string, error) {
+	return "2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6", nil // Mock with a well-known private key from test test ... test junk derivation (9).
```

Comment:
from configuration file?

Reply:
Yes all the hardcoded values for now are hardcoded but they will be fetch from configuration in the future.
For now, will keep the constant and hardcoded values and will be removed in the PR 3.